### PR TITLE
feat: Global Search and Pagination

### PR DESCRIPTION
**Pull request by arteevraina**
This Pull Request : 

- [x] Implements Global Search.
- [x] Separate reducer and actions for search functionality.
- [x] Better error handling.
- [x] Separate Reusable component for Package Item.
- [x] Improve the package search API to send total Pages.
- [x] Implemented Pagination in Frontend.
- [x] Separate reusable component for Pagination tiles. 
- [x] Add sorting options in frontend.

### CODE REVIEW

File: registry/src/store/actions/searchActions.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import axios from "axios";
+
+export const SEARCH_PACKAGE = "SEARCH_PACKAGE";
+export const SEARCH_SUCCESS = "SEARCH_SUCCESS";
+export const SEARCH_FAILURE = "SEARCH_FAILURE";
+
+export const searchPackage = (query, page) => async (dispatch) => {
+  const url = `${
+    process.env.REACT_APP_REGISTRY_API_URL
+  }/packages?query=${encodeURIComponent(query)}&page=${encodeURIComponent(
```
Comment (henilp105) [inline registry/src/store/actions/searchActions.js]: @arteevraina  as We have implemented searching by  various parameters ,I think we should allow the user to search the packages by those parameters.

File: flask/packages.py [hunk 2]
```diff
@@ -33,6 +34,24 @@ def search_packages():
     )
     page = int(page) if page else 0
 
+    packages_per_page = 10
+
+    # Count the number of documents in the database related to query.
+    total_documents = db.packages.count_documents({
```
Comment (henilp105) [inline flask/packages.py]: can you shift this down after the query has been formatted. for similar outputs to the queries.

File: registry/src/store/actions/searchActions.js (hunk 1 above)
Comment (arteevraina) [reply registry/src/store/actions/searchActions.js]: Isn't this block of code already doing that in the API https://github.com/henilp105/registry/blob/main/flask/packages.py#L40 ?
@henilp105

Comment (henilp105) [reply registry/src/store/actions/searchActions.js]: I am referring to give the user of the frontend the options to sort by the given parameters,  ( we would have to provide the options on frontend to search by name , and other in parameters dictionary).

File: registry/src/store/actions/searchActions.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      sorted_by = "updatedat";
+    } else {
+      sorted_by = "";
+    }
+
+    const url = `${
+      process.env.REACT_APP_REGISTRY_API_URL
+    }/packages?query=${encodeURIComponent(query)}&page=${encodeURIComponent(
+      page
+    )}&sorted_by=${encodeURIComponent(sorted_by)}`;
```
Comment (henilp105) [inline registry/src/store/actions/searchActions.js:20]: @arteevraina as we have implemented the functionality of sortby and  sort with ascending and descending , we could also provide options for that in UI. (Like sort by date and ascending or descending).

**Comment by henilp105** [comment]
Looks good. Thanks @arteevraina .

**Pull request merged**